RevenueCat/purchases-kmp
# Refactor expect/actual from more enums to avoid typealiases

### Description
This continues the work in #137 

In this one, we are removing them from the enums:
- PackageType
- VerificationResult
- BillingFeature
- GoogleReplacementMode
- RecurrenceMode

Additionally, there are a couple extra changes that were made as part of this PR:
- Use `ReplacementMode` in the purchase methods.
- Change `PricingPhase` to use extensions to avoid having to rewrite the whole offering response classes in this PR. I expect we will want to revert this change in future PRs, but did this in order to keep it clean for now.
